hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -264,6 +251,146 @@ public void renameTable(TableIdentifier from, TableIdentifier originalTo) {
     }
   }
 
+  @Override
+  public boolean viewExists(TableIdentifier identifier) {
+    return HiveCatalogUtil.isTableWithTypeExists(clients, identifier, TableType.VIRTUAL_VIEW);
+  }
+
+  @Override
+  public boolean dropView(TableIdentifier identifier) {
+    if (!isValidIdentifier(identifier)) {
+      return false;
+    }
+    try {
+      String database = identifier.namespace().level(0);
+      String viewName = identifier.name();
+      Table table = clients.run(client -> client.getTable(database, viewName));
+      HiveCatalogUtil.validateTableIsIcebergView(table, fullTableName(name, identifier));
+      clients.run(
+          client -> {
+            client.dropTable(database, viewName);
+            return null;
+          });
+      LOG.info("Dropped View: {}", identifier);
+      return true;
+
+    } catch (NoSuchViewException | NoSuchObjectException e) {
+      LOG.info("Skipping drop, View does not exist: {}", identifier, e);
+      return false;
+    } catch (TException e) {
+      throw new RuntimeException("Failed to drop " + identifier, e);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to dropView", e);
+    }
+  }
+
+  @Override
+  public List<TableIdentifier> listViews(Namespace namespace) {
+    try {
+      return listContents(namespace, TableType.VIRTUAL_VIEW.name(), icebergPredicate());
+    } catch (UnknownDBException e) {
+      throw new NoSuchNamespaceException("Namespace does not exist: %s", namespace);
+
+    } catch (TException e) {
+      throw new RuntimeException("Failed to list all views under namespace " + namespace, e);
+
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new RuntimeException("Interrupted in call to listViews", e);
+    }
+  }
+
+  private List<TableIdentifier> listContents(
+      Namespace namespace, String tableType, Predicate<Table> tablePredicate)
+      throws TException, InterruptedException {
+    Preconditions.checkArgument(
+        isValidateNamespace(namespace), "Missing database in namespace: %s", namespace);
+    String database = namespace.level(0);
+    List<String> tableNames =
+        StringUtils.isNotEmpty(tableType)
+            ? clients.run(client -> client.getTables(database, "*", TableType.valueOf(tableType)))
+            : clients.run(client -> client.getAllTables(database));
+    List<Table> tableObjects =
+        clients.run(client -> client.getTableObjectsByName(database, tableNames));
```

Comment:
This HMS call was not used before the patch when all tables were listed. After the patch we fetch all of the tables, even if it is not neccessary.

I remember cases where there were too many tables in a database, and fetching all of the tables were problematic, so I would like us to find a better solution here

Reply:
I have added offset based `[loading](https://github.com/apache/iceberg/pull/8907/files#diff-3fdd21d335b598f5354de287be6a8830d78e381c89274f0d6bf85c4c9966b8f0R342)` of HMS table. PTAL.

Reply:
When we were optimizing for performance, every extra HMS call is problematic. I would prefer a solution, where we do not call the HMS if it is not neccessary